### ISSUE TITLE
Minify problems

### ISSUE THREAD

**Issue by Integral** [substance/substance#1116]
There is a problems with minifying library.
For instance [this function](https://github.com/substance/substance/blob/develop/model/DocumentChange.js#L44)
turns to
```
_extractInformation(t) {
  function e(e) {
    switch (e.type) {
    case "create":
    case "delete":
      {
        let t = e.val;
        t.hasOwnProperty("start") && (updated[t.start.path] = !0), t.hasOwnProperty("end") && (updated[t.end.path] = !0);
        break
      }
    case "update":
    case "set":
      {
        let n = t.get(e.path[0]);
        n && (n.isPropertyAnnotation() ? updated[n.start.path] = !0 : n.isContainerAnnotation() && affectedContainerAnnos.push(n));
        break
      }
    default:
      throw new Error("Illegal state")
    }
  }
  if (!this._extracted) {
    let n = this.ops, i = {}, r = {}, o = {}, s = [];
    for (let a = 0; a < n.length; a++) {
      let t = n[a];
      "create" === t.type && (i[t.val.id] = t.val, delete r[t.val.id]), "delete" === t.type && (delete i[t.val.id], r[t.val.id] = t.val), "set" !== t.type && "update" !== t.type || (o[t.path] = !0, o[t.path[0]] = !0), e(t)
    }
    s.forEach(function(e) {
      let n = t.get(e.containerId, "strict"), i = n.getPosition(e.start.path[0]), s = n.getPosition(e.end.path[0]);
      for (let a = i; a <= s; a++) {
        let t, e = n.getChildAt(a);
        t = e.isText() ? [e.id, "content"] : [e.id], r[e.id] || (o[t] = !0)
      }
    }), Object.keys(r).length > 0 && _n(o, function(t, e) {
      let n = e.split(",")[0];
      r[n] && delete o[e]
    }), this.created = i, this.deleted = r, this.updated = o, this._extracted = !0
  }
}
```
Which will error as there is no `updated` variable. There are might be an option in uglify to avoid this kind of things, dunno, but it looks like we can't pass any minify option to our bundler.

**Comment by Integral** [comment]
PR: #1117.

**Comment by obuchtala** [comment]
This is a weird bug.
@Integral Is this still happening with the current version? now we use `uglify-es@^3.3.9`

**Comment by Integral** [comment]
@oliver---- I can't see uglify dependency in develop or minify-fixes branch and in bundler repo as well. Also it looks like we don't have bundler tasks for minifying or it is not clear. And we didn't describe this feature anywhere. Do you need any help with that? I can take over.

**Comment by obuchtala** [comment]
It seems that we only use it here: https://github.com/substance/substance/blob/develop/make.js#L131
But this is only vendor bundling, not dist bundling.

I thought this is a problem with our bundling. But it seems that this issue was reported from minifying projects that use substance.
@Integral As you are the creator of this issue, could you just check if the problem still exists in the context where you have observed this?

**Comment by obuchtala** [comment]
There could be a good chance that this issue has resolved on `uglify` side. 
Because the 'problematic' code is absolutely fine. And rather 'fixing' our correct code I would like to understand the source of the problem, and maybe there is no problem anymore.

**Comment by Integral** [comment]
I can't remember... But I can try to make a sample project. But why don't we wan't to support minifying on this (substance) level?

**Comment by obuchtala** [comment]
You can, but only for the final build, i.e. only for `node make publish`
During dev uglify makes builds too slow.

**Comment by Integral** [comment]
Do we want to change uglify-js-harmony requirement [here](https://github.com/substance/bundler/blob/master/src/commands/MinifyCommand.js#L57)?

**Comment by obuchtala** [comment]
Yeah, let me change that. 
Some hours ago I have published a new bundler version which has an option for `b.js()`
```
b.js('index.es.js', { 
  ...,
  minify: true
})
```
I need to that new version on substance. There might be regressions.

**Comment by obuchtala** [comment]
I have published substance-bundler@0.24.1.
For substance bundling, `b.minify()` is the better option than `b.js(..., {... minify:true})` because we want an uncompressed bundle, too.

**Comment by obuchtala** [comment]
Obsolete.
